File: src/main/java/com/thunderbear06/ai/task/tasks/EntityBasedTask.java
package com.thunderbear06.ai.task.tasks;

import com.thunderbear06.ai.task.Task;
import com.thunderbear06.entity.android.AndroidEntity;
import net.minecraft.world.entity.LivingEntity;

public abstract class EntityBasedTask extends Task {
    private final LivingEntity target;

    protected EntityBasedTask(AndroidEntity android, LivingEntity target) {
        super(android);
        this.target = target;
    }

    @Override
    public boolean shouldTick() {
        return target.isAlive();
    }

    @Override
    public void tick() {
        android.getLookControl().setLookAt(target);
    }

    protected boolean isInRange(double distance) {
        return android.distanceToSqr(target) <= distance * distance;
    }

    protected LivingEntity getTarget() {
        return target;
    }
}

File: src/main/kotlin/com/thunderbear06/ai/task/tasks/EntityBasedTask.kt
package com.thunderbear06.ai.task.tasks

import com.thunderbear06.ai.task.Task
import com.thunderbear06.entity.android.AndroidEntity
import net.minecraft.world.entity.LivingEntity

abstract class EntityBasedTask(android: AndroidEntity, @JvmField protected val target: LivingEntity) : Task(android) {
    override fun shouldTick(): Boolean = target.isAlive

    override fun tick() {
        android.lookControl.setLookAt(target)
    }

    protected fun isInRange(distance: Double): Boolean = android.distanceToSqr(target) <= distance * distance
}
